### ISSUE TITLE
Trouble reproducing results on the paper

### ISSUE THREAD

**Issue by aanchan** [Uppaal/lm-ood#1]
Hi @Uppaal thanks for the amazing work. We are trying to reproduce some of your results on our local machine and we are unable to replicate the numbers in the paper in Table 2.

- Conda environment was created with python 3.9.0. 
- Package versions in original “requirements.txt” were removed due to the installer unable to resolve conflict among package requirements. 
- All packages were installed successfully after the removal of versions.
- Config files were modified to use target dataset:
For each set of ood detection experiment being run, in domain data is set to “task_name”  and included in the “ood_datasets” list. The real ood data is included as the other element of the list.
Example config used: 
```
{
    "MODEL_DIR": "~/Desktop/lm-ood/model",
    "DATA_DIR": "~/Desktop/lm-ood/data",
    "root_path": "~/Desktop/lm-ood/data/src",
    "task_name": "imdb",
    "ood_datasets": "['imdb','sst2']",
    "model_class": "roberta",
    "do_train": false,
    "num_train_epochs": 10,
    "CUDA_VISIBLE_DEVICES": 0
}
```
The rest parameters were left without being modified in config.py. The default config as is, we assume uses SupCon as Loss Function. (seed not changed)
 
In the code, some of the paths [are hardcoded to local paths](https://github.com/Uppaal/lm-ood/blob/e17ca5d6ac5a9c554db2f7d3fcaaaab8c9bb9373/src/utils/model_utils.py#L28), so auto download breaks. So a pre-trained Roberta model was downloaded from hugging face https://huggingface.co/roberta-base and placed under model/pretained_models folder named as “roberta_base”.
After run 
```
. bin/start.sh
 
followed by  python run run_ood_detection.py
numbers were recorded from the log as spreadsheet.
To run training with supcon, we modified the “do_train” variable to true and re-run the script.
```

Our results comparing those from the paper are below:
20NG -> SST2
--
<img width="1146" alt="Screen Shot 2023-09-28 at 10 53 30 AM" src="https://github.com/Uppaal/lm-ood/assets/3464128/b7a10683-263a-49ba-8dee-f6e46f95ba99">
<img width="1730" alt="Screen Shot 2023-09-28 at 10 54 53 AM" src="https://github.com/Uppaal/lm-ood/assets/3464128/f812866a-db71-4a89-82d6-0116d25b163d">

20NG ->RTE
<img width="1062" alt="Screen Shot 2023-09-28 at 10 56 28 AM" src="https://github.com/Uppaal/lm-ood/assets/3464128/87c957cb-7e9a-4f7d-b05f-28fad5809948">
<img width="1357" alt="Screen Shot 2023-09-28 at 10 57 55 AM" src="https://github.com/Uppaal/lm-ood/assets/3464128/d956ed97-5020-43c1-9a51-65dbd7ffe732">

ImDB-> SST2
<img width="1097" alt="Screen Shot 2023-09-28 at 10 59 43 AM" src="https://github.com/Uppaal/lm-ood/assets/3464128/471261e5-4c72-4da5-80b0-345f88404d4f">
<img width="1357" alt="Screen Shot 2023-09-28 at 10 59 15 AM" src="https://github.com/Uppaal/lm-ood/assets/3464128/c4fb2646-cfdd-4668-882b-14ad6eb5f978">

Furthermore, when use_train=True we find that the supcon results are different on different machines, note that we have left the seed unchanged.

Any help to let us reproduce the results would be greatly appreciated.
